packages/compiler/core/checker.ts

```diff
@@ -2573,6 +2580,44 @@ export function createChecker(program: Program): Checker {
     });
   }
 
+  function checkEnumSpreadMember(
+    parentEnum: EnumType,
+    targetNode: TypeReferenceNode,
+    existingMemberNames: Set<string>
+  ): EnumMemberType[] {
+    const members: EnumMemberType[] = [];
+    const targetType = getTypeForNode(targetNode);
+
+    if (!isErrorType(targetType)) {
+      if (targetType.kind !== "Enum") {
+        program.reportDiagnostic(createDiagnostic({ code: "spread-enum", target: targetNode }));
+        return members;
+      }
+
+      for (const member of targetType.members) {
+        if (existingMemberNames.has(member.name)) {
+          program.reportDiagnostic(
+            createDiagnostic({
+              code: "enum-member-duplicate",
+              format: { name: member.name },
+              target: targetNode,
+            })
+          );
+        } else {
+          existingMemberNames.add(member.name);
+          const clonedMember = cloneType(member, {
+            enum: parentEnum,
```

Comment:
We might want to have a `sourceMember` like we have `sourceProperty` for spread properties